Classes/Common/SolrSearchResult/ResultDocument.php

```diff
@@ -0,0 +1,259 @@
+<?php
+
+/**
+ * (c) Kitodo. Key to digital objects e.V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo and TYPO3 projects.
+ *
+ * @license GNU General Public License version 3 or later.
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ */
+
+namespace Kitodo\Dlf\Common\SolrSearchResult;
+
+use Kitodo\Dlf\Common\SolrSearchResult\Highlight;
+use Kitodo\Dlf\Common\SolrSearchResult\Page;
+use Kitodo\Dlf\Common\SolrSearchResult\Region;
+
+/**
+ * ResultDocument class for the 'dlf' extension. It keeps te result of the search in the SOLR index.
+ *
+ * @author Beatrycze Volk <beatrycze.volk@slub-dresden.de>
+ * @package TYPO3
+ * @subpackage dlf
+ * @access public
+ */
+class ResultDocument {
+
+    /**
+     * The identifier
+     *
+     * @var string
+     * @access private
+     */
+    private $id;
+
+    /**
+     * The unified identifier
+     *
+     * @var string|null
+     * @access private
+     */
+    private $uid;
+
+    /**
+     * The page on which result was found
+     *
+     * @var int
+     * @access private
+     */
+    private $page;
+
+    /**
+     * All snippets imploded to one string
+     *
+     * @var string
+     * @access private
+     */
+    private $snippets;
+
+    /**
+     * All pages in which search phrase was found
+     *
+     * @var array(Page)
+     * @access private
+     */
+    private $pages = [];
+
+    /**
+     * All regions in which search phrase was found
+     *
+     * @var array(Region)
+     * @access private
+     */
+    private $regions = [];
+
+    /**
+     * All highlights of search phrase
+     *
+     * @var array(Highlight)
+     * @access private
+     */
+    private $highlights = [];
+
+    /**
+     * The snippets for given record
+     *
+     * @var array
+     * @access private
+     */
+    private $snippetsForRecord = [];
+
+    /**
+     * The constructor for result.
+     *
+     * @access public
+     *
+     * @param array $record: Array of found document record
+     * @param array $highlighting: Array of found highlight elements
+     * @param array $fields: Array of fields used for search
+     *
+     * @return void
+     */
+    public function __construct($record, $highlighting, $fields) {
+        $this->id = $record[$fields['id']];
+        $this->uid = $record[$fields['uid']];
+        $this->page = $record[$fields['page']];
+
+        $highlightingForRecord = $highlighting[$record[$fields['id']]][$fields['fulltext']];
+        $this->snippetsForRecord = $highlightingForRecord['snippets'];
```

Comment:
This maybe null and the getArrayByIndex() function will fail late.
```suggestion
        $this->snippetsForRecord = is_array($highlightingForRecord['snippets']) ? $highlightingForRecord['snippets'] : [];
```

Reply:
Shouldn't this ternary operator have something between ?: or is it something specific to is_array and it doesn't need assignment?

Reply:
I think, you're right. This would return "true" in case of a valid array. I've updated my suggestion.